sysdeps/managarm/generic/file.cpp

```diff
@@ -3917,6 +3917,57 @@ int sys_pread(int fd, void *buf, size_t n, off_t off, ssize_t *bytes_read) {
 	}
 }
 
+int sys_pwrite(int fd, const void *buf, size_t n, off_t off, ssize_t *bytes_written) {
+	SignalGuard sguard;
+
+	auto handle = getHandleForFd(fd);
+	if (!handle)
+		return EBADF;
+
+	managarm::fs::CntRequest<MemoryAllocator> req(getSysdepsAllocator());
+	req.set_req_type(managarm::fs::CntReqType::PT_PWRITE);
+	req.set_fd(fd);
+	req.set_size(n);
+	req.set_offset(off);
+
+	frg::string<MemoryAllocator> ser(getSysdepsAllocator());
+	req.SerializeToString(&ser);
+
+	auto [offer, send_req, imbue_creds, send_data, recv_resp] = exchangeMsgsSync(
+		handle,
+		helix_ng::offer(
+			helix_ng::sendBuffer(ser.data(), ser.size()),
```

Comment:
Why don't you use ``helix_ng::sendBragiHeadTail`` here?

Reply:
I was under the impression that only works for the new bragi messages? This is still a legacy CntRequest.

Reply:
Ah, if it's still a legacy request then this is fine, yes.

Reply:
Wait is it? Afaict in ``protocols/fs/`` there is only an ``fs.bragi``.

Reply:
Well it’s part of the big CntRequest thingy. That is legacy no? I assumed the bragi sendHeadTail thing only worked for dedicated request messages?

Reply:
No it works fine for this too.